soci/soci_index.go

```diff
@@ -57,26 +56,47 @@ var (
 	errNotLayerType = errors.New("not a layer mediaType")
 )
 
-// nolint:revive
-type SociIndex struct {
-	MediaType    string `json:"mediaType"`
-	ArtifactType string `json:"artifactType"`
-	// descriptors of ztocs
-	Blobs []ocispec.Descriptor `json:"blobs,omitempty"`
-	// descriptor of image manifest
-	Subject ocispec.Descriptor `json:"subject,omitempty"`
+/// Index represents an interface to interact with ORAS/OCI Ref Types.
+/// In the fullness of time, ORAS support will be removed from the snapshotter
+/// and this interface will not be needed.
+type Index interface {
+	/// Returns the media type of the artifact manifest.
+	GetMediaType() string
 
-	Annotations map[string]string `json:"annotations,omitempty"`
+	/// Returns the artifact type of the manifest.
+	GetArtifactType() string
+
+	/// Returns the blobs contained in the manifest.
+	GetBlobs() []ocispec.Descriptor
+
+	/// Returns an optional reference to an existing manifest.
+	/// For OCI Reference Types, it returns the Refers field.
+	/// For ORAS, it returns the Subject field.
+	GetRefers() *ocispec.Descriptor
+
+	/// Returns the manifest annotations.
+	GetAnnotations() map[string]string
+
+	/// Converts the index to a JSON byte array.
+	ToJSON() ([]byte, error)
```

Comment:
nit: this suggestion is idiomatic go
```suggestion
	JSON() ([]byte,error)
```
https://go.dev/doc/effective_go#interface-names

Reply:
Sure, I'll update it.